### PR TITLE
feat(framework,medusa): Ensure publishable key middleware is set for all store endpoints

### PR DESCRIPTION
**Pull request by riqwan**
what:

- publishable key middleware is applied to all store routes on a router level

RESOLVES CC-519

### CODE REVIEW

File: packages/core/modules-sdk/src/remote-query/parse-filters.ts [hunk 1]
```diff
@@ -33,23 +29,23 @@ export function parseAndAssignFilters(
   const joinerConfigs = MedusaModule.getAllJoinerConfigs()
 
   for (const [filterKey, filterValue] of Object.entries(filters)) {
-    let entryAlias!: JoinerServiceConfigAlias
+    // let entryAlias!: JoinerServiceConfigAlias
```
Comment (riqwan) [inline packages/core/modules-sdk/src/remote-query/parse-filters.ts]: These changes will be cleaned up in Adrien's PR

File: packages/core/modules-sdk/src/remote-query/parse-filters.ts [hunk 2]
```diff
@@ -33,23 +29,23 @@ export function parseAndAssignFilters(
...
+    // entryAlias = alias
     entryJoinerConfig = joinerConfig
 
-    const entryEntity = entitiesMap[entryAlias.entity!]
-    if (!entryEntity) {
-      throw new Error(
-        `Entity ${entryAlias.entity} not found in the public schema of the joiner config from ${entryJoinerConfig.serviceName}`
-      )
-    }
+    // const entryEntity = entitiesMap[entryAlias.entity!]
```
Comment (sradevski) [inline packages/core/modules-sdk/src/remote-query/parse-filters.ts]: Is this intentional?

Comment (riqwan) [reply packages/core/modules-sdk/src/remote-query/parse-filters.ts]: yes, the latest rebase from develop will have a comment in there

File: packages/core/framework/src/http/middlewares/ensure-publishable-api-key.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  if (!isPresent(publishableApiKey)) {
+    try {
+      throw new MedusaError(
+        MedusaError.Types.NOT_ALLOWED,
+        `Publishable API key required in the request header: ${PUBLISHABLE_KEY_HEADER}. You can manage your keys in settings in the dashboard.`
+      )
+    } catch (e) {
+      return next(e)
+    }
```
Comment (adrien2p) [inline packages/core/framework/src/http/middlewares/ensure-publishable-api-key.ts:30]: Q: Do you remember why we are not doing 😂 
```ts
const error = new MedusaError(...)
return next(error)
```

**Review by adrien2p — APPROVED** [review]

File: packages/core/framework/src/http/middlewares/ensure-publishable-api-key.ts (hunk 3 above)
Comment (riqwan) [reply packages/core/framework/src/http/middlewares/ensure-publishable-api-key.ts:30]: I tried this and the middleware got stuck 🤷🏻

**Pull request merged**